js/gdm/authPrompt.js

```diff
@@ -39,6 +41,14 @@ var BeginRequestType = {
     DONT_PROVIDE_USERNAME: 1
 };
 
+function _getMachineId() {
+    let machineId = getFileContents('/etc/machine-id') ||
```

Comment:
If you adjust this to use `Shell.get_file_contents_utf8_sync` you'll need to use `try { } catch { }` for robustness.

Reply:
Done.